src/analysis/souffle/tags.dl

```diff
@@ -0,0 +1,41 @@
+//-----------------------------------------------------------------------------
```

Comment:
@markww If you want to work on supporting confidentiality and integrity, it could be useful for you to take a look at the [Flume](https://pdos.csail.mit.edu/papers/flume-sosp07.pdf) paper. The label model we would use here is pretty similar to that one because it is based on pairs of sets. (The flowsto, meet, and join will work about the same for us as they do in that paper. Downgrading using capabilities and other aspects of their label model are not things we will pick up).

Another label model we have talked about often is [FLAM](https://www.cs.cornell.edu/andru/papers/flam/flam-csf15.pdf), which is also quite nice -- but I think it's more useful to look at Flume first because what we will implement will really be closer to Flume. Flume and FLAM actually have a lot in common (as do all IFC label models essentially) once you notice FLAM's normal form has a confidentiality and integrity component (and that all labels can be reduced to normal form).

We can't actually represent FLAM in datalog because checking flowsto in FLAM is reducible to the SAT problem because for both of its components it uses full (rather than semi-) lattices. Flume *is* implementable in datalog because its label components use just semi-lattices. 

(It's possible that if we had our own datalog solver we could approximate FLAM by using the hypercube label model, but this is slow with datalog's bottom-up solver --- let's save that for the distant future).